SearchDropdown update

mzgajner opened the pull request:
This PR implements [this ticket](https://trello.com/c/YtIcYARM/26-updejt-dropdownov).

As I started working on it, I figured it deserves a refactor, so i did the following:

- Switch to standard update behaviour (value prop + input event) so that we don't have to modify a parent prop like savages.
- Move default placeholder to component (it was repeated in many components)
- Consistently import as PSearchDropdown
- Use events instead of function props for select and clear callbacks

Apart from that, the added functionality is:
- Dropdown items may now include `image` (string with url of image) or `color` (string with CSS-valid color value) property. Adding one of these toggles items to large (double height) mode that uses styling from the linked ticket above and displays an image circle or a color dot.
- When the dropdown is filtered by entering some text, results are highlighted in bold.

mzgajner committed:
Refactor SearchDropdown

- Switch to standard update behaviour (value prop + input event)
- Move default placeholder to component
- Consistently import as PSearchDropdown
- Use events instead of function props for select and clear callbacks

mzgajner committed:
Style group labels

mzgajner commented on cards/p/seznam-poslancev/data.json:
```diff
@@ -8734,73 +8734,6 @@
     },
-    {
-      "person": {
-        "name": "Sun\u010dana Glavak",
```
This lady made it impossible to load the card - @LepkoQQ am I missing something?

LepkoQQ replied:
Yes, she doesn't seem to have any data (see all the nulls). So it probably tries to load something there.

mzgajner replied:
Yeah, I get that's why it fails, what I'm wondering is - am I missing some configuration that would allow me to open this card or is it possible that the stuff in data.json is bad?

muki replied:
maybe the problem is that you have slovenian sources set up at one place and croatian sources someplace else. check config. it looks to me like your data.json is croatian - where did you get it from?

mzgajner committed:
Use new color variables

mzgajner committed:
Implement two variants of large dropdown items

- First one is image circle (add 'image' prop with url to item)
- Second one is color dot (add 'color' prop with hex to item)
- Adjust keyboard navigation
- Extract css class generation to a method

mzgajner committed:
Implement filter result highlighting
